### PR TITLE
Add fast EDT

### PR DESCRIPTION
**Pull request by vovaf709**

### CODE REVIEW

File: imops/morphology.py [hunk 1]
```diff
@@ -1,8 +1,17 @@
-from typing import Callable
+from typing import Callable, Tuple, Union
 from warnings import warn
 
 import numpy as np
-from scipy.ndimage import generate_binary_structure
+from edt import edt
+from scipy.ndimage import distance_transform_edt as scipy_distance_transform_edt, generate_binary_structure
+from scipy.ndimage._nd_image import euclidean_feature_transform
```
Comment (maxme1) [inline imops/morphology.py:7]: unsafe import. need a fallback at least

File: imops/morphology.py [hunk 2]
```diff
@@ -354,3 +363,152 @@ def binary_opening(
...
+        return scipy_distance_transform_edt(input, sampling, return_distances, return_indices)
+
+    if input.ndim > 3:
+        warn("Fast Euclidean Distance Transform is only supported for ndim<=3. Falling back to scipy's implementation.")
+        return scipy_distance_transform_edt(input, sampling, return_distances, return_indices)
+
+    if (not return_distances) and (not return_indices):
+        raise RuntimeError('At least one of `return_distances`/`return_indices` must be True')
+
+    input = np.atleast_1d(np.where(input, 1, 0).astype(np.int8))
```
Comment (maxme1) [inline imops/morphology.py:488]: mb should just ban the case with ndim == 0?

File: imops/morphology.py [hunk 3]
```diff
@@ -354,3 +363,152 @@ def binary_opening(
     """
 
     return _binary_opening(image, footprint, output, boxed, num_threads, backend)
+
+
+def distance_transform_edt(
+    input: np.ndarray,
+    sampling: Tuple[float] = None,
+    return_distances: bool = True,
+    return_indices: bool = False,
```
[the remark this answers is not in the document]
Comment (vovaf709) [reply imops/morphology.py:372]: It throws `RuntimeError` like in `scipy`

File: imops/morphology.py (hunk 2 above)
Comment (vovaf709) [reply imops/morphology.py:488]: I just copypasted this from `scipy`, it works ok for ndim == 0 despite it has no sence

File: _build_utils.py [hunk 4]
```diff
@@ -78,9 +78,9 @@ def get_ext_modules():
             [f'{name}/src/_{prefix}{module}.pyx'],
             include_dirs=[LazyImport('numpy')],
             library_dirs=[NumpyLibImport()] if module in modules_to_link_against_numpy_core_math_lib else [],
-            libraries=['npymath'] + ['m'] * (not on_windows)
-            if module in modules_to_link_against_numpy_core_math_lib
-            else [],
+            libraries=(
```
[the remark this answers is not in the document]
Comment (vovaf709) [reply _build_utils.py:81]: Didn't get it, u want to unwrap this to clear if-else... ?

File: imops/morphology.py (hunk 1 above)
Comment (vovaf709) [reply imops/morphology.py:7]: For python 3.6-3.12 it works ok. And I don't know what fallback to use

File: imops/morphology.py (hunk 2 above)
Comment (maxme1) [reply imops/morphology.py:488]: fair enough

File: imops/morphology.py (hunk 1 above)
Comment (maxme1) [reply imops/morphology.py:7]: ok. let's wait until this changes. then we'll patch it

**Review by maxme1 — APPROVED** [review]

**Pull request merged**